modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/AddressResourceTest.java

```diff
@@ -14,15 +14,275 @@
+import com.liferay.commerce.service.CommerceOrderLocalService;
+import com.liferay.commerce.test.util.CommerceTestUtil;
+import com.liferay.headless.commerce.delivery.cart.client.dto.v1_0.Address;
+import com.liferay.portal.kernel.model.Country;
+import com.liferay.portal.kernel.model.Region;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.CountryLocalService;
+import com.liferay.portal.kernel.service.RegionLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalServiceUtil;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.test.rule.Inject;
 
-import org.junit.Ignore;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.runner.RunWith;
 
 /**
  * @author Andrea Sbarra
  */
-@Ignore
 @RunWith(Arquillian.class)
 public class AddressResourceTest extends BaseAddressResourceTestCase {
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_user = UserTestUtil.addUser(testCompany);
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			testCompany.getCompanyId(), testGroup.getGroupId(),
+			_user.getUserId());
+
+		_commerceAccount =
+			_commerceAccountLocalService.addBusinessCommerceAccount(
+				"Test Business Account", 0, null, null, true, null, null, null,
+				_serviceContext);
+
+		_commerceCurrency = CommerceCurrencyTestUtil.addCommerceCurrency(
+			testGroup.getCompanyId());
+
+		_commerceChannel = CommerceTestUtil.addCommerceChannel(
+			testGroup.getGroupId(), _commerceCurrency.getCode());
+
+		_country = _countryLocalService.fetchCountryByNumber(
+			_serviceContext.getCompanyId(), "000");
+
+		if (_country == null) {
+			_country = _countryLocalService.addCountry(
+				"ZZ", "ZZZ", true, true, null, RandomTestUtil.randomString(),
+				"000", RandomTestUtil.randomDouble(), true, false, false,
+				_serviceContext);
+
+			_region = _regionLocalService.addRegion(
+				_country.getCountryId(), true, RandomTestUtil.randomString(),
+				RandomTestUtil.randomDouble(), "ZZ", _serviceContext);
+		}
+		else {
+			_region = _regionLocalService.getRegion(
+				_country.getCountryId(), "ZZ");
+		}
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+
+		if (_commerceOrder != null) {
+			_commerceOrderLocalService.deleteCommerceOrder(_commerceOrder);
+		}
+
+		if (_commerceAddress != null) {
+			_commerceAddressLocalService.deleteCommerceAddress(
+				_commerceAddress);
+		}
+
+		if (_commerceAccount != null) {
+			_commerceAccountLocalService.deleteCommerceAccount(
+				_commerceAccount);
+		}
+
+		if (_user != null) {
+			UserLocalServiceUtil.deleteUser(_user);
```

Comment:
Don't we trust @DeleteAfterTestRun annotation?
We should use that where possible.
(This is for all the entities used)

Reply:
you are right, i will made a changes